File: src/main/java/com/github/morningzeng/toolset/ui/TabSupport.java
package com.github.morningzeng.toolset.ui;

import com.intellij.openapi.project.Project;

import javax.swing.Icon;
import javax.swing.JComponent;
import javax.swing.JTabbedPane;

/**
 * @author Morning Zeng
 * @since 2024-05-09
 */
public interface TabSupport {

    String title();

    Icon icon();

    String tips();

    JComponent component(final Project project);

    default void putTab(final Project project, JTabbedPane tabbedPane) {
        tabbedPane.addTab(this.title(), this.icon(), this.component(project), this.tips());
    }

}

File: src/main/kotlin/com/github/morningzeng/toolbox/ui/TabSupport.kt
package com.github.morningzeng.toolbox.ui

import com.intellij.openapi.wm.ToolWindow
import javax.swing.Icon
import javax.swing.JComponent
import javax.swing.JTabbedPane

/**
 * @author Morning Zeng
 * @since 2025-05-15
 */
interface TabSupport {

    val title: String

    val icon: Icon?

    val tips: String

    fun component(toolWindow: ToolWindow): JComponent


    fun putTab(tabbedPane: JTabbedPane, toolWindow: ToolWindow) {
        tabbedPane.addTab(this.title, this.icon, this.component(toolWindow), this.tips)
    }
}